test/common/quic/active_quic_listener_test.cc

```diff
@@ -603,14 +603,23 @@ TEST_P(ActiveQuicListenerTest, EcnReportingIsEnabled) {
   Network::Socket& socket = ActiveQuicListenerPeer::socket(*quic_listener_);
   absl::optional<Network::Address::IpVersion> version = socket.ipVersion();
   EXPECT_TRUE(version.has_value());
-  int optval;
+  int optval = 0;
   socklen_t optlen = sizeof(optval);
   Api::SysCallIntResult rv;
   if (*version == Network::Address::IpVersion::v6) {
     rv = socket.getSocketOption(IPPROTO_IPV6, IPV6_RECVTCLASS, &optval, &optlen);
-  } else {
-    rv = socket.getSocketOption(IPPROTO_IP, IP_RECVTOS, &optval, &optlen);
+    EXPECT_EQ(rv.return_value_, 0);
+    EXPECT_EQ(optval, 1);
+    rv = socket.getSocketOption(IPPROTO_IPV6, IPV6_V6ONLY, &optval, &optlen);
+    EXPECT_EQ(rv.return_value_, 0);
+    if (optval) {
```

Comment:
Since we EXPECT_EQ'd optval, let's remove this conditional. Though do we need and ifdef or some sort 'cause doesn't iOS do things differently with respect to the socket options that need to be set?